sanic/request.py

```diff
@@ -134,6 +136,26 @@ def __bool__(self):
             return True
         return False
 
+    def get(self, key, default=None):
+        """Request data storage. Arbitrary per-request data may be stored."""
```

Comment:
I think the `request.get(...)` /  `request[...]` modes of access should be deprecated, if people are on board with that lets add a warning in the docstrings

Reply:
Agreed. If we want to let user add additional request context but not let them use the request object directly as a `dict`, then we are better off moving away from the the `get(k, default)`